test/test-time-profiler.ts

```diff
@@ -0,0 +1,89 @@
+import * as delay from 'delay';
+import {perftools} from '../src/profile';
+import {TimeProfiler} from '../src/profilers/time-profiler';
+import {getIndexOrAdd} from '../src/util';
+
+let assert = require('assert');
+
+describe('TimeProfiler', function() {
+  describe('profile', function() {
+    it('should profile during duration and finish profiling after duration',
+       function() {
+         const duration = 500;
+         const sampleInterval = 1000;
+         let profiler = new TimeProfiler(sampleInterval);
+         let profilePromise = profiler.profile(duration);
```

Comment:
These tests actually enable/disable the profiler, so these would be hard to get running in parallel.

Can you look into whether you can either dependency inject or mock the time profiler bindings so that these become true unit tests?

I think you would still want tests that actually enable profiling, but I would consider them 'system tests'. The unit tests should focus on unit behaviors.